opensrp-anc/src/main/java/org/smartregister/anc/library/model/ContactVisit.java

```diff
@@ -142,6 +149,57 @@ private void updateEventAndRequiredStepsField(String baseEntityId, PartialContac
         }
     }
 
+    private void processTasks(JSONObject formObject) {
+        try {
+            String encounterType = formObject.getString(ConstantsUtils.JsonFormKeyUtils.ENCOUNTER_TYPE);
+            if (formObject.has(ConstantsUtils.JsonFormKeyUtils.ENCOUNTER_TYPE) && StringUtils.isNotBlank(encounterType) && ConstantsUtils.JsonFormUtils.ANC_TEST_ENCOUNTER_TYPE.equals(encounterType)) {
+                JSONObject dueStep = formObject.optJSONObject(JsonFormConstants.STEP1);
+                if (dueStep != null && dueStep.has(JsonFormConstants.STEP_TITLE) && "Due".equals(dueStep.getString(JsonFormConstants.STEP_TITLE))) {
+                    JSONArray stepFields = dueStep.optJSONArray(JsonFormConstants.FIELDS);
+                    if (stepFields != null && stepFields.length() > 0) {
+                        for (int i = 0; i < stepFields.length(); i++) {
+                            JSONObject field = stepFields.getJSONObject(i);
+                            if (field != null && field.has(JsonFormConstants.IS_VISIBLE) && field.getBoolean(JsonFormConstants.IS_VISIBLE)) {
+                                JSONArray jsonArray = field.optJSONArray(JsonFormConstants.VALUE);
+                                if (jsonArray == null || !(jsonArray.length() > 0)) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use opposite operator instead of the logic complement operator.](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=4809639)